pkg/clustermesh/clustermesh/clustermesh.go

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package clustermesh
```

Comment:
Yeah, I also personally find `pkg/clustermesh/clustermesh` to be not that great. I'm not sure about a better one though. One possibility could be to just keep everything in the base clustermesh package, losing a bit of separation (and have the kvstoremesh stuff in `pkg/clustermesh/kvstoremesh`). Otherwise I could do the other way round and move the generic bits in `pkg/clustermesh/internal` or `pkg/clustermesh/generic` (is this what you were proposing?). WDYT?

Reply:
I've moved the generic part to the `pkg/clustermesh/internal` package, and restored back the business logic in `pkg/clustermesh`, adapting variables names accordingly. I've additionally extracted a few clusterconfig-related utilities in a `pkg/clustermesh/utils` package, as they are also used by the clustermesh-apiserver.